website/blog/2022-07-12-change-data-capture-metrics.md

```diff
@@ -0,0 +1,348 @@
+
+```sql
+{{
+    config(
+        materialized='incremental'
+    )
+}}
+```
+
+By materializing `int_income_history` as incremental but *not* including a `unique_key` config, dbt will only execute `INSERT` statements – new rows will be added, but old rows will remain unchanged.
+
+The rest of `int_income_history` would look like this:
+
+```sql
+...
+
+select
+    *
+from {{ ref('fct_income') }}
+{% if is_incremental() %}
+    where true
+{% endif %}
+```
+
+There are a few additional configs that Joanne might find helpful:
+
+- she can use the `on_schema_change` config to handle schema changes if  new columns are added and/or deleted from `fct_income`
+- she can also set the `full_refresh` config to false in order to prevent accidental loss of the historical data 
+- she can build this table in a custom `schema` if she wants to enforce specific role-based permissions for this historical table
+- she can specify a time-grain `unique_key` if she wants to reduce the amount of versions being captured 
+    - for example, if she only wants to capture the final version of each day she could set `unique_key = date_trunc('day', run_timestamp)`. This is excluded from the example below, as we are making the assumption that Joanne does indeed want to capture every version of every version of `fct_income`.
+
+The final config block for `int_income_history` might look something like this:
+
+```sql
+{{
+    config(
+        materialized='incremental',
+	  full_refresh=false,
+	  schema='history',
+	  on_schema_change='sync_all_columns'
+    )
+}}
+```
+
+As a final step, Joanne would create `fct_income_history` to join in the seed file to determine which versions are “correct”. Her new DAG looks like this, where `int_income_history` is an incremental model without a unique key:
+
+![](/img/blog/2022-07-12-change-data-capture-metrics/int-income-history-dag.png)
+
+The final output of `fct_income_history` would look identical to `stg_snapshot_fct_income` from her initial approach: 
+
+| month_year | product_category | revenue | run_timestamp | correct_version |
+|:---:|:---:|:---:|:---:|:---:|
+| January 2022 | clothing | 100 | 01/30/22 12:00:00 | FALSE |
+| January 2022 | electronics | 200 | 01/30/22 12:00:00 | FALSE |
+| January 2022 | books | 300 | 01/30/22 12:00:00 | FALSE |
+| January 2022 | clothing | 50 | 02/03/22 16:00:00 | FALSE |
+| January 2022 | electronics | 150 | 02/03/22 16:00:00 | FALSE |
+| January 2022 | books | 200 | 02/03/22 16:00:00 | FALSE |
+| January 2022 | clothing | 52 | 02/10/22 08:00:00 | TRUE |
+| January 2022 | electronics | 152 | 02/10/22 08:00:00 | TRUE |
+| January 2022 | books | 202 | 02/10/22 08:00:00 | TRUE |
+
+## Solution #2: Upstream Snapshots
+
+Alternatively, Joanne could snapshot her source data and add flexibility to her modeling so that all historical versions are calculated `at the same time`. Let’s look at our example.
```

Comment:
```suggestion
Alternatively, Joanne could snapshot her source data and add flexibility to her modeling so that all historical versions are calculated *at the same time*. Let’s look at our example.
```